src/qibo/tests_new/test_core_circuit_qasm.py

```diff
@@ -0,0 +1,173 @@
+"""Tests executing Qibo circuits created from OpenQASM code."""
+import pytest
+import numpy as np
+import cirq
+from qibo import gates
+from qibo.models import Circuit
+from cirq.contrib.qasm_import import circuit_from_qasm, exception
+
+
+# Absolute testing tolerance for cirq-qibo comparison
+_atol = 1e-7
+
+
+def test_from_qasm_simple(backend, accelerators):
+    target = f"""OPENQASM 2.0;
+include "qelib1.inc";
+qreg q[5];
+h q[0];
+h q[1];
+h q[2];
+h q[3];
+h q[4];"""
+    import qibo
+    original_backend = qibo.get_backend()
+    qibo.set_backend(backend)
+    c = Circuit.from_qasm(target, accelerators)
+    assert c.nqubits == 5
+    assert c.depth == 1
+    for i, gate in enumerate(c.queue):
+        assert gate.__class__.__name__ == "H"
+        assert gate.qubits == (i,)
+    target_state = np.ones(32) / np.sqrt(32)
+    np.testing.assert_allclose(c(), target_state)
+    qibo.set_backend(original_backend)
+
+
+def test_simple_cirq():
+    c1 = Circuit(2)
+    c1.add(gates.H(0))
+    c1.add(gates.H(1))
+    final_state_c1 = c1()
+
+    c2 = circuit_from_qasm(c1.to_qasm())
+    c2depth = len(cirq.Circuit(c2.all_operations()))
+    assert c1.depth == c2depth
+    final_state_c2 = cirq.Simulator().simulate(c2).final_state

```

Comment:
I think we should replace `final_state` with `final_state_vector` in all places.